miprometheus/models/mac_sequential/model.py

```diff
@@ -439,98 +442,202 @@ def plot(self, data_dict, logits, sample=0):
+                    # preprocess attention image, reshape
+                    attention_size = int(np.sqrt(attention_mask.size(-1)))
+
+                    # attention mask has size [batch_size x 1 x(H*W)]
+                    attention_mask = attention_mask.view(-1, 1, attention_size, attention_size)
+
+                    # upsample attention mask
+                    m = torch.nn.Upsample(
+                        size=[width, height], mode='bilinear', align_corners=True)
+                    up_sample_attention_mask = m(attention_mask)
+                    attention_mask = up_sample_attention_mask[sample, 0]
+
+                    # upsample attention mask
+                    original_grid_size=7
+                    preds_pointing = preds_pointing.view(images.size(0),images.size(1),original_grid_size, -1)
+                    mm =torch.nn.Upsample(size=[width, height] , mode= 'bilinear')
+                    up_sample_preds_pointing = mm(preds_pointing)
+                    up_sample_preds_pointing = up_sample_preds_pointing[sample][i]
+
+                    # preprocess question, pick one sample number
+                    attention_question = attention_question[sample]
+
+                    # Create "Artists" drawing data on "ImageAxes".
+                    num_artists = len(fig.axes) + 1
+                    artists = [None] * num_artists
+
+                    # set title labels
+                    ax_image.set_title(
+                        'COG image:')
+                    ax_attention_question.set_xticklabels(
+                        ['h'] + words, rotation='vertical', fontsize=10)
+                    ax_step.axis('off')
+                    ax_attention_image.set_title(
+                        'Pointing Distribution:')
+
+                    # Tell artists what to do:
+                    artists[0] = ax_image.imshow(
+                        image, interpolation='nearest', aspect='auto')
+                    artists[1] = ax_attention_image.imshow(
+                        image, interpolation='nearest', aspect='auto')
+                    artists[2] = ax_attention_image.imshow(
+                        up_sample_preds_pointing.detach().numpy(),
+                        interpolation='nearest',
+                        aspect='auto',
+                        alpha=0.5,
+                        cmap='Blues')
+                    artists[3] = ax_attention_question.imshow(
+                        attention_question.transpose(1, 0),
+                        interpolation='nearest', aspect='auto', cmap='Reds')
+                    artists[4] = ax_step.text(
+                        0, 0.5, 'Reasoning step index: ' + str(
+                            step) + ' | Question type: ' + tasks,
+                        fontsize=15)
+
+                    # Add "frames" to artist list
+                    frames.append(artists)
+
+            # Plot figure and list of frames.
+            self.plotWindow.update(fig, frames)
```

Comment:
I have noticed that you are not returning information if window was closed or not.

       return self.plotWindow.is_closed

This is crucial, as workers use that information for breaking the training (if user presses exit button).

Have you tested that?

Reply:
ok good